### PR TITLE
Wizard input fields should be a form

### PR DESCRIPTION
**Pull request by gitstart**
### DEMO VIDEO

https://www.loom.com/share/4896a7ba0b6d4c30965825183efb6c98

fixes https://github.com/amplication/amplication/issues/5818

---
This code was written and reviewed by GitStart Community. Growing great engineers, one PR at a time.

### CODE REVIEW

**Comment by yuval-hazaz** [comment]
Thanks for this PR! 

I appreciate the efforts, but I must say I do not like this type of solution where we catch the key down separately on each page
Did you try to think of a more generic solution where the default button in the form is the "Continue" button, and the rest of the logic, including validation is untouched ?

**Review by yuval-hazaz — CHANGES_REQUESTED** [review]

File: packages/amplication-client/src/Resource/create-resource/wizard-pages/CreateServiceName.tsx [hunk 1]
```diff
@@ -1,18 +1,35 @@
...
+          formik.values.serviceName?.trim() !== "" &&
+          !Object.keys(formik.errors).length
+        ) {
+          goNextPage && goNextPage();
+        }
+      }
+    },
+    [formik, goNextPage]
+  );
+
```
[the remark this answers is not in the document]
Comment (jaysoni-ash42) [reply packages/amplication-client/src/Resource/create-resource/wizard-pages/CreateServiceName.tsx:32]: @yuval-hazaz For each step the keyDown event react's differently for input keyDown submits the form whereas for others it updates a state then we move to the continue using tab and hit enter then it submits.

[the remark this answers is not in the document]
Comment (jaysoni-ash42) [reply packages/amplication-client/src/Resource/create-resource/wizard-pages/CreateServiceName.tsx:32]: @yuval-hazaz  how about using a custom keyEvent wrapper like this 

```
import { ENTER } from "../../../util/hotkeys";

const withKeyboardEvents = (WrappedComponent) => {
  const WithKeyboardEvents = (props) => {
    const handleKeyDown = async (event) => {
      const { key, target } = event;
      if (key === ENTER) {
        if (target.tagName === "INPUT") {
          if (props.formik.validateForm && props.goNextPage) {
            await props.formik.validateForm();
            if (!Object.keys(props.formik.errors).length) {
              props.goNextPage();
            }
          }
        }
        if (target.tagName === "DIV") {
          target.click();
        }
      }
    };

    return <WrappedComponent {...props} onKeyDown={handleKeyDown} />;
  };

  return WithKeyboardEvents;
};

export default withKeyboardEvents;
```
and using this through out the serviceWizard and handling accordingly

[the remark this answers is not in the document]
Comment (yuval-hazaz) [reply packages/amplication-client/src/Resource/create-resource/wizard-pages/CreateServiceName.tsx:32]: thanks @jaysoni-ash42 
I will look into it and get back to you ASAP

[the remark this answers is not in the document]
Comment (yuval-hazaz) [reply packages/amplication-client/src/Resource/create-resource/wizard-pages/CreateServiceName.tsx:32]: @jaysoni-ash42 after some playing around with this issue, I came up with a new solution that is based on triggering a click on the button (which already incorporates all the logic about validation, etc)
please have a look at this PR I created and let me know what you think
https://github.com/amplication/amplication/pull/6386

Although your solution looks valid, I didn't know how to validate it, so I tried to play with it myself, which eventually brought me to this solution. Thank you so much for the time and effort you spent on this PR.

[the remark this answers is not in the document]
Comment (jaysoni-ash42) [reply packages/amplication-client/src/Resource/create-resource/wizard-pages/CreateServiceName.tsx:32]: Appreciate @yuval-hazaz  will definitely check out the new pr.

**Pull request closed without merging**